Java/jni/native_glue.cc

```diff
@@ -68,12 +70,15 @@ namespace litecore {
             assert(env != nullptr);
             if (js != nullptr) {
                 jboolean isCopy;
+                _jstr = js;
+                _env = env;
+
                 const char *cstr = env->GetStringUTFChars(js, &isCopy);
                 if (!cstr)
                     return; // Would it be better to throw an exception?
                 _slice = slice(cstr);
-                _jstr = js;
-                _env = env;
+
+                fixBrokenModifiedUTF8(cstr);
```

Comment:
Should we #ifdef this to make it Android-only?

Reply:
If possible, I guess.  I am not sure if this works on Java desktop or not so if we support that someday it would need to be rechecked (from the linked bug it seems that the NDK is just lagging behind and that desktop JRE fixed this issue, but my lack of Java expertise makes me unsure).